src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -123,18 +121,28 @@ private static bool DoesAnyParentHaveAPropertyWithDefaultValue(CodeClass current
             currentClass.Name = refinedClassName;
         }
         else if (current is CodeProperty currentProperty &&
-                propertyNames &&
+                classNames &&
                 refineName(currentProperty.Type.Name) is string refinedPropertyTypeName &&
                 !currentProperty.Type.Name.Equals(refinedPropertyTypeName))
         {
             currentProperty.Type.Name = refinedPropertyTypeName;
         }
         else if (current is CodeMethod currentMethod &&
-                methodNames &&
-                refineName(currentMethod.ReturnType.Name) is string refinedMethodTypeName &&
-                !currentMethod.ReturnType.Name.Equals(refinedMethodTypeName))
+                classNames)
         {
-            currentMethod.ReturnType.Name = refinedMethodTypeName;
+            foreach (var param in currentMethod.Parameters)
+            {
+                if (refineName(param.Type.Name) is string refinedTypeName &&
+                    !param.Type.Name.Equals(refinedTypeName))
```

Comment:
```suggestion
                    !param.Type.Name.Equals(refinedTypeName, StringComparison.OrdinalIgnoreCase))
```

Reply:
I'm fixing this by using `StringComparison.Ordinal` everywhere in this method, pretty convinced that we should be checking for the casing here.